src/Features/Core/Portable/SymbolSearch/SymbolSearchUpdateNoOpEngine.cs

```diff
@@ -13,6 +13,11 @@ internal sealed class SymbolSearchUpdateNoOpEngine : ISymbolSearchUpdateEngine
 {
     public static readonly SymbolSearchUpdateNoOpEngine Instance = new();
 
+    public void Dispose()
+    {
+        // Nothing to do for the no-op version.
+    }
```

Comment:
Note: the actual Dispose we care about *already* exists, but is currently not called through any codepaths.  It is here: https://github.com/dotnet/roslyn/blob/9b6dd878170636c4a7771aae644aede20123213b/src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngineProxy.cs#L16

What thsi PR does is hook things up to ensure that it is called.  To ensure that we actually dispose the RemoteServiceConnection instance we're holding onto.